File: Python/6py 2/split_strings.py
"""
|--------------------------------------------------------------------------
| Split Strings
|--------------------------------------------------------------------------
|
"""

def solution(s):
   res = []
   for i in range(0, len(s), 2):
      x = s[i:i+2]
      if len(x) < 2: x += "_"
      res.append(x)
   return res


from re import findall
def solution(s):
   return findall(".{2}", s + "_")


def solution(s):
   return [(s + "_")[i:i + 2] for i in range(0, len(s), 2)]


print(solution("abcdef"), ["ab", "cd", "ef"])
print(solution("abcdefg"), ["ab", "cd", "ef", "g_"])
print(solution(""), [])

File: PHP/6php 2/split_strings.php
<?php
/*
|--------------------------------------------------------------------------
| Split Strings
|--------------------------------------------------------------------------
|
*/

function solution($s) {
   $res = [];
   for ($i = 0; $i < strlen($s); $i += 2) {
      $x = substr($s, $i, 2);
      if (strlen($x) < 2) $x .= "_";
      array_push($res, $x);
   }
   return $res;
}


function solution($str): array {
   preg_match_all('/\w{2}/', $str . '_', $matches);
   return array_values($matches[0]);
}


function solution($str) {
   return array_slice(str_split("{$str}__", 2), 0, -1);
}


print_r(solution("abcdef"));
print_r(solution("abcdefg"));
print_r(solution(""));